library/std/src/sys/unix/io.rs

```diff
@@ -74,3 +75,23 @@ impl<'a> IoSliceMut<'a> {
         unsafe { slice::from_raw_parts_mut(self.vec.iov_base as *mut u8, self.vec.iov_len) }
     }
 }
+
+#[unstable(feature = "is_terminal", issue = "80937")]
+impl io::IsTerminal for sys::stdio::Stdin {
+    fn is_terminal() -> bool {
+        unsafe { libc::isatty(libc::STDIN_FILENO) != 0 }
```

Comment:
Won't this return true in case of an error?

Reply:
The man page indicates that this function can only return 0 or 1: https://linux.die.net/man/3/isatty

I could make it `libc::isatty(libc::STDOUT_FILENO) == 1` to be more explicit?